### ISSUE TITLE
Suggestions and 2 bugs (one causes crashes)

### ISSUE THREAD

**Issue by Icouldntthinkofone** [Huzuni/Huzuni-1.11#17]
using: 1.10.2-forge1.10.2-12.18.3.2185
Whenever I turn on Projectiles it crashes. No other mods installed.

In MobESP, giant slimes show up as having 1x1 hitbox (only happens sometimes)

Suggestions:
Please make Journeymap work. It seems to map fine and the fullscreen version works, but I can't see the actual mini-map

The text renderer looks a bit blurred??

Please add nametag health for mobs

Add horse stats functionality under nametag (speed, jump and health)

Make it so you can have tracers drawn to players only and not mobs, and make it so you can have different tracer colors (hues) for different entity types

In the modules screen add new tabs for everything any other mod tries to render. Sorta of a GUI manager.

Add an option to disable Huzuni's GUI manager and just let every mod render what it wants how it wants on the screen

PS: I've been using Huzuni for quite a while since the early versions, and I really like the idea of making it a Forge mod since I don't have to sacrifice other dank mods in order to cheat. Keep up the good work!

PSS: when the client crashes from projectiles, I can't open the crash log via the minecraft launcher for some reason

**Comment by Icouldntthinkofone** [comment]
PSSS: Do you still update the 1.10 version? If not please do, some servers haven't updated yet.

**Comment by brudin** [comment]
JourneyMap will be fixed for the next update!
Horse stats is a really good idea, as are the mob tags, we'll definitely look into that.
You can have tracers point to players only, look at the settings for it ;) Different colors is a good idea
"In the modules screen add new tabs for everything any other mod tries to render. Sorta of a GUI manager" - Not really sure what you mean
"Add an option to disable Huzuni's GUI manager and just let every mod render what it wants how it wants on the screen" If you go to the widget settings, you can disable the other stuff if that's what you mean
We definitely need to rework the Xray configuration, I don't really like how we have the preset thing and prefer how we had it back in the day where it would just show all blocks.
The projectiles crash bug is fixed for the next release

The plan right now is moving all the mod code away from Minecraft and instead to our own wrapper so that we can update all versions (1.10, 1.11, 1.8, etc.) at the same time.  Setting this all up is taking a bit of time but once it's done all versions will be able to be updated simultaneously.

**Comment by Icouldntthinkofone** [comment]
>You can have tracers point to players only, look at the settings for it

Point to players only AND draw boxes around mobs? didn't see any options for that tbh, I'll have another look.

>Not really sure what you mean

My current guess so far, is that Huzuni blocks off other mods' attempts at drawing elements on the HUD (The minimap doesn't draw the actual minimap, etc), what I'm suggesting is that in the modules screen you create additional entries for other things in the HUD other mods try to render. So for example, if a mod displays something, say text in the upper-left corner and it doesn't have an option to disable it I can simply go into Huzuni's module tab and disable it off there

Basically a HUD manager.

>The projectiles crash bug is fixed for the next release

10/10, when is it coming out?

>The plan right now is moving all the mod code away from Minecraft and instead to our own wrapper so that we can update all versions (1.10, 1.11, 1.8, etc.)

I wonder why no one did this before. This is genius tbh.

**Comment by brudin** [comment]
> Point to players only AND draw boxes around mobs? didn't see any options for that tbh, I'll have another look.

Ooooh, now I get what you mean.  The Players/Mobs/Animals options just toggle whatever you have enabled (lines/esp) for that entity.  Right now you can't just have it draw lines to players and only boxes for mobs.

> [...] Basically a HUD manager

You can disable/move all of the HUD elements already, check out the widgets tab.  The text displaying is just from the notifications widget, which can also be disabled.  I think having a 'Show Notifications' option for mods that display notifications would be nice, though.

> 10/10, when is it coming out?

Once the wrapper is done, it'll be with that release.  The 'hard part' is done, now we just need to think of how we want to handle rendering (for simple things like lines, if we want to continue to use Minecraft's or just use our own), and we need to rewrite the screens for things like the alt manager so that it doesn't rely on Minecraft code.  I'm hoping within the next couple of weeks it'll be ready, Hal and I have both been pretty busy with other stuff this past week (as you can probably tell by the git commits, lol)

**Comment by Icouldntthinkofone** [comment]
>You can disable/move all of the HUD elements already, 

Even HUD elements from other mods? That's what I'm talking about. 

>if we want to continue to use Minecraft's or just use our own)

imo, choose the option with less lag and less memory leaking.

**Comment by brudin** [comment]
Oooh, like Forge mods?  Not sure if that would be easy to do since a lot of Forge mods use HUD render events for other things too.

Performance is definitely the main focus with our choice!
